info-compiler/src/main/java/com/google/sps/data/NewsArticle.java

```diff
@@ -14,28 +14,43 @@
 
 package com.google.sps.data;
 
+import com.google.sps.webcrawler.WebCrawler;
+import java.util.Date;
+
 /** A news article, including its metadata and content. */
 public class NewsArticle {
+  private static final int LOWEST_PRIORITY = WebCrawler.CUSTOM_SEARCH_RESULT_COUNT;
   private String title;
   private String url;
   private String content;
   private String abbreviatedContent;
   private String summarizedContent;
+  private String publisher;
+  private Date publishedDate;
+  // A value in range [1, {@code LOWEREST_PRIORITY].
+  private int priority;
 
-  public NewsArticle(String title, String url, String content) {
-    this.title = (title == null) ? "" : title;
+  /**
+   * Consturcts a {@code NewsArticle}. If {@code publishedDate) is null, initialize with the
+   * standard base time: January 1, 1970, 00:00:00 GMT.
+   */
+  public NewsArticle(String url, String publisher, Date publishedDate, int priority) {
     this.url = (url == null) ? "" : url;
-    this.content = (content == null) ? "" : content;
-    this.abbreviatedContent = null;
-    this.summarizedContent = null;
+    this.publisher = (publisher == null) ? "" : publisher;
+    this.publishedDate = (publishedDate == null) ? new Date(0) : publishedDate;
+    this.priority = (priority > LOWEST_PRIORITY) ? LOWEST_PRIORITY : priority;
```

Comment:
Don't set this.prioritiy twice.

Reply:
Thanks for pointing this out! I'll change it